packages/eas-cli/src/uploads.ts

```diff
@@ -31,6 +32,37 @@ export async function uploadWithPresignedPostAsync(
   file: string,
   presignedPost: PresignedPost,
   handleProgressEvent?: ProgressHandler
+): Promise<Response> {
+  return await promiseRetry(
+    async retry => {
+      let response: Response;
+      try {
+        response = await uploadWithPresignedPostInternalAsync(
+          file,
+          presignedPost,
+          handleProgressEvent
+        );
+      } catch (e: any) {
+        return retry(e);
+      }
+
+      if (!response.ok) {
```

Comment:
(Comment was lost) Requesting changes to look at the response status and retry only on 408, 429, and 5xx errors. I don't think it's useful to retry a 401 for example.